### PR TITLE
adds dpr transformation 

### PR DESCRIPTION
**Pull request by selul**
Following the conversation from #92 I've tried to implement the DPR transformation as well as support this in client hints. 

The DPR factor can be a float number between 1 and 8 and will multiply the resulting image sizes by this factor. 
 
Looking forward to your feedback, @DarthSim

### CODE REVIEW

File: process.go [hunk 1]
```diff
@@ -245,7 +245,19 @@ func processImage(ctx context.Context) ([]byte, error) {
 	checkTimeout(ctx)
 
 	imgWidth, imgHeight, angle, flip := extractMeta(img)
+	if po.Dpr > 1 {
+		if po.Width != 0 {
+			po.Width = int(float32(po.Width) * po.Dpr)
+		} else {
+			po.Width = int(float32(imgWidth) * po.Dpr)
```
Comment (DarthSim) [inline process.go]: This breaks the behavior. Zero width/height should stay zero.

File: processing_options.go [hunk 2]
```diff
@@ -470,6 +472,19 @@ func applyBlurOption(po *processingOptions, args []string) error {
 	return nil
 }
 
+func applyDprOption(po *processingOptions, args []string) error {
+	if len(args) > 1 {
+		return fmt.Errorf("Invalid dpr arguments: %v", args)
+	}
+
+	if d, err := strconv.ParseFloat(args[0], 32); err == nil || (d >= 1 || d <= 8) {
```
Comment (DarthSim) [inline processing_options.go]: I don't think we need the upper limit here. If a user wants to generate an image with 100500 DPR, well, as we use to say in Russia, he is his own evil Pinocchio. I think this limit is reasonable only for DPR header.

Comment (selul) [reply processing_options.go]: hehe, makes sense.

File: process.go (hunk 1 above)
Comment (selul) [reply process.go]: Thanks for the feedback, but what about when the users do not request any width/height in specific, and he just wants to get an image with dpr=2 for instance. where is the right location to apply this?

**Comment by selul** [comment]
Hey, 

I was thinking about the DPR transformation something that should apply to the resulting image, indifferent of what kind of transformation was done before.  
Furthermore, if the image is cropped to 200x100 after the cropping is done and the resulting image is ready, we should rescale the image according to the DPR factor. Looking at some cloud-based image services, it seems they follow the same logic: 
https://docs.imgix.com/apis/url/dpr 

Thinking more about this, I might not have added the dpr logic in the proper place in `process.go` ? What do you think?

**Comment by DarthSim** [comment]
Yeah, I think we need more complex behavior for DPI. Let's keep it as is in this PR (with fixes), and I'll think how to make it properly.

File: processing_options.go (hunk 2 above)
Comment (DarthSim) [reply processing_options.go]: Also, i think it's a good idea to support LowDPI, so let's allow dpi < 1 and > 0.

**Comment by DarthSim** [comment]
Thanks for the contribution again!

**Pull request merged**